### ISSUE TITLE
IOS only showing images and HRs and not text

### ISSUE THREAD

**Issue by JustWoody** [cesarvr/pdf-generator#118]
This is probably me, but this plugin was working brilliantly, showing the html and css that I was passing to it in the PDF

Then due to the impending cliff edge with Apple of removing UIWebView I made a load of changes to my project to remove any links to UIWebView and since then this plugin is no longer outputting any text that is in my local html

Not sure if this plugin is reliant on any framework or library that I have accidentally removed

Any help would be appreciated + sorry if this is not the location for this

**Comment by cesarvr** [comment]
At the moment the plugin is using [WKWebView](https://github.com/cesarvr/pdf-generator/blob/master/src/ios/BNHtmlPdfKit.m#L49), so it should work fine without legacy WebKit.

The library don't use any dependencies aside from WebKit, have you [updated to the last version](https://www.npmjs.com/package/cordova-pdf-generator) ? I removed some UIWebView dependencies.

**Comment by JustWoody** [comment]
Yes, as soon as I noticed the problem I removed and re-added the plugin

**Comment by cesarvr** [comment]
Also now that I see (again) the title of the issue, your problem could be also that the HTML you are providing doesn't include the text. 

To test this try to print the payload your are passing to the plugin to the actual document.

You can try something like: 

```js
/*
 Assuming you are generating the payload with a bunch of HTML
*/

document.write(payload)
//The point here is to print the content into the browser to see your check if the text is appearing. 
//The plugin should display the same thing.

```

**Comment by JustWoody** [comment]
Already did that too and in the log everything is in the HTML string including the text (thanks you for trying so far)

**Comment by JustWoody** [comment]
Does it rely on coreGraphics as I removed that?

**Comment by cesarvr** [comment]
Nop, the only dependency is WebKit which the plugin use to render the PDF. 

> Already did that too and in the log everything is in the HTML string including the text (thank you for trying so far)

If I was you, I would try to render the PDF content in the browser (not print in a console), another way to test this is to copy paste the logs on a HTML file and open it in the browser.  

I know it was working before, but is something you might consider because at the end this plugin is just a [WYSIWYG](https://en.wikipedia.org/wiki/WYSIWYG) with respect to the browser. 

Cheers.

**Comment by JustWoody** [comment]
I already also output the HTML to the browser as this is to allow the user to save what appears on the screen
